native/cocos/audio/apple/AudioEngine-inl.mm

```diff
@@ -125,20 +125,25 @@ - (void)handleInterruption:(NSNotification *)notification {
             isAudioSessionInterrupted = false;
 
             NSError *error = nil;
-            [[AVAudioSession sharedInstance] setActive:YES error:&error];
+            if([AVAudioSession sharedInstance] respondsToSelector:@selector(setActive)){
+                [[AVAudioSession sharedInstance] setActive:YES error:&error];
+            }
             alcMakeContextCurrent(s_ALContext);
 
             if ([UIApplication sharedApplication].applicationState != UIApplicationStateActive) {
                 ALOGD("AVAudioSessionInterruptionTypeEnded, application != UIApplicationStateActive, resumeOnBecomingActive = true");
                 resumeOnBecomingActive = true;
             }
+            else{
+                pauseOnResignActive = false;
+            }
         }
     } else if ([notification.name isEqualToString:UIApplicationWillResignActiveNotification]) {
         ALOGD("UIApplicationWillResignActiveNotification");
         if (pauseOnResignActive) {
             pauseOnResignActive = false;
             ALOGD("UIApplicationWillResignActiveNotification, alcMakeContextCurrent(nullptr)");
-            alcMakeContextCurrent(nullptr);
+            alcMakeContextCurrent(s_ALContext);
```

Comment:
Audio doesn`t need to play at this state. In addition, we also need to consider the impact of this change on WeChat voice calls.

Reply:
> Audio doesn`t need to play at this state. In addition, we also need to consider the impact of this change on WeChat voice calls.
in ios system 17.5.1 when pickup the phone, and comeback , the game audio will be silence. just because of at this state, ios system will trigger audio, but context has been set nullptr. you can have a try.

Reply:
> Audio doesn`t need to play at this state. In addition, we also need to consider the impact of this change on WeChat voice calls.

in ios system 17.5.1 when pickup the phone, and comeback , the game audio will be silence. just because of at this state, ios system will trigger audio, but context has been set nullptr. you can have a try.

Reply:
Myabe can remove the code:   

> alcMakeContextCurrent(nullptr);

Reply:
> Myabe can remove the code:
> 
> > alcMakeContextCurrent(nullptr);

I will have a try